### PR TITLE
Remove `git_root` from Commands

### PR DESCRIPTION
**Pull request by waydegg**
*This PR is part of a series of PRs to remove `git_root` from Mentat, and allow Sessions to be run anywhere on the filesystem.*

### CODE REVIEW

File: mentat/commands.py [hunk 1]
```diff
@@ -279,9 +277,12 @@ async def apply(self, *args: str) -> None:
             return
 
         for i, (feature, score) in enumerate(results, start=1):
-            label = feature.ref()
-            if label.startswith(str(git_root)):
-                label = label[len(str(git_root)) + 1 :]
+            if feature.path.is_relative_to(session_context.cwd):
+                label = f"{feature.path.relative_to(session_context.cwd)}"
```
Comment (granawkins) [inline mentat/commands.py:281]: Could feature.ref have an optional 'cwd' arg? It's quite useful elsewhere for quickly grabbing/comparing the path/interval.

**Review by granawkins — CHANGES_REQUESTED** [review]

**Review by PCSwingle — APPROVED** [review]

File: mentat/commands.py (hunk 1 above)
Comment (waydegg) [reply mentat/commands.py:281]: What would the `cwd` arg do in this case?

Comment (granawkins) [reply mentat/commands.py:281]:
```
    def ref(self, cwd: Optional[Path] = None):
        path_string = str(self.path) 
        if cwd and self.path.is_relative_to(cwd):
            path_string = self.path.relative_to(cwd)
        if self.level == CodeMessageLevel.INTERVAL:
            interval_string = f"{self.interval.start}-{self.interval.end}"
            return f"{path_string}:{interval_string}"
        return path_string

```

File: tests/commands_test.py [hunk 2]
```diff
@@ -43,6 +43,7 @@ async def test_commit_command(
     assert subprocess.check_output(["git", "status", "-s"], text=True) == ""
 
 
+# TODO: test without git
```
Comment (biobootloader) [inline tests/commands_test.py:46]: these test changes coming in another PR?

**Review by biobootloader — APPROVED** [review]

File: tests/commands_test.py (hunk 2 above)
Comment (waydegg) [reply tests/commands_test.py:46]: Yup! The Session and SessionContext still depend on git_root, so we can't use the non-git testbed in this PR

Comment (waydegg) [reply tests/commands_test.py:46]: Yup! The Session and SessionContext still depend on git_root, so we can't use the non-git testbed in this PR

**Comment by jakethekoenig** [comment]
Is there a reason this PR isn't merged? Seems better to merge as we go to avoid a bunch of merge conflicts when you're done with the series.

**Review by granawkins — APPROVED** [review]

**Comment by waydegg** [comment]
>Is there a reason this PR isn't merged? Seems better to merge as we go to avoid a bunch of merge conflicts when you're done with the series.

Yeah this could have been merged sooner

**Pull request merged**